src/compiler/transformers/declarations.ts

```diff
@@ -1157,13 +1157,20 @@ namespace ts {
                         fakespace.locals = createSymbolTable(props);
                         fakespace.symbol = props[0].parent!;
                         const declarations = mapDefined(props, p => {
-                            if (!isPropertyAccessExpression(p.valueDeclaration)) {
-                                return undefined; // TODO GH#33569: Handle element access expressions that created late bound names (rather than silently omitting them)
+                            if (!isPropertyAccessExpression(p.valueDeclaration) && !isElementAccessExpression(p.valueDeclaration) && !isBinaryExpression(p.valueDeclaration)) {
+                                return undefined;
+                            }
+                            if (hasDynamicName(p.valueDeclaration) && !resolver.isLateBound(getParseTreeNode(p.valueDeclaration) as Declaration)) {
+                                return undefined;
+                            }
+                            const name = unescapeLeadingUnderscores(p.escapedName);
+                            if (!isIdentifierText(name, ScriptTarget.ES3)) {
+                                return undefined; // TODO: Rather than quietly eliding (as is current behavior), maybe we should issue errors?

```

Comment:
Issuing errors would be a breaking change, since our past behavior with late bound stuff has always been to elide what we couldn't emit. Still, it means we're quietly omitting type information from the declaration file. Usually that'd warrant an error in any other scenario, imo.